flask/todo_app/extensions/principal_ext.py

```diff
@@ -0,0 +1,8 @@
+from flask_principal import Principal
```

Comment:
I really don't like having the directory information duplicated in the filename.  There has to be a better way to solve the problem.